backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecApiImpl.scala

```diff
@@ -477,7 +497,12 @@ class SparkPlanExecApiImpl extends SparkPlanExecApi {
    * @return
    */
   override def genExtendedColumnarPostRules(): List[SparkSession => Rule[SparkPlan]] = {
-    List(spark => NativeWritePostRule(spark))
+    val Array(major, minor, _) = SparkShimLoader.getSparkShims.getShimDescriptor.toString.split('.')
+    if (major.toInt > 3 || (major.toInt == 3 && (minor.toInt >= 4))) {
+      List()
+    } else {
+      List(spark => NativeWritePostRule(spark))
+    }
```

Comment:
Can we somehow do this in shim? Would that require for lot of effort?

Reply:
@zhztheplayer  Good catch. I will add this in shim in the following PRs. Thanks.